src/all/mangareaderto/src/eu/kanade/tachiyomi/extension/all/mangareaderto/MangaReader.kt

```diff
@@ -0,0 +1,187 @@
+package eu.kanade.tachiyomi.extension.all.mangareaderto
+
+import android.app.Application
+import android.net.Uri
+import androidx.preference.ListPreference
+import androidx.preference.PreferenceScreen
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.asObservableSuccess
+import eu.kanade.tachiyomi.source.ConfigurableSource
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import org.json.JSONObject
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+
+open class MangaReader(
+    override val lang: String
+) : ConfigurableSource, ParsedHttpSource() {
+    override val name = "MangaReader"
+
+    override val baseUrl = "https://mangareader.to"
+
+    override val supportsLatest = true
+
+    override val client = network.client.newBuilder()
+        .addInterceptor(MangaReaderImageInterceptor())
+        .build()
+
+    override fun latestUpdatesRequest(page: Int) =
+        GET("$baseUrl/filter?sort=latest-updated&language=$lang&page=$page", headers)
+
+    override fun latestUpdatesSelector() = searchMangaSelector()
+
+    override fun latestUpdatesNextPageSelector() = searchMangaNextPageSelector()
+
+    override fun latestUpdatesFromElement(element: Element) =
+        searchMangaFromElement(element)
+
+    override fun popularMangaRequest(page: Int) =
+        GET("$baseUrl/filter?sort=most-viewed&language=$lang&page=$page", headers)
+
+    override fun popularMangaSelector() = searchMangaSelector()
+
+    override fun popularMangaNextPageSelector() = searchMangaNextPageSelector()
+
+    override fun popularMangaFromElement(element: Element) =
+        searchMangaFromElement(element)
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList) =
+        if (query.isNotBlank()) {
+            GET("$baseUrl/search?keyword=$query&page=$page", headers)
```

Comment:
It would be better to build the URL by using a dedicated class such as `HttpUrl` or `Uri`, since the user can type special characters and break the URL in the way it is now (simple concatenation).

```suggestion
            Uri.parse("$baseUrl/search").buildUpon().run {
                appendQueryParameter("keyword", query)
                appendQueryParameter("page", page.toString())
                
                GET(toString(), headers)
            }
```